crates/hir-def/src/nameres/collector.rs

```diff
@@ -1233,6 +1229,15 @@ impl DefCollector<'_> {
                     );
                     let loc: MacroCallLoc = self.db.lookup_intern_macro_call(call_id);
 
+                    if !self.db.enable_proc_attr_macros() {
+                        self.def_map.diagnostics.push(DefDiagnostic::unresolved_proc_macro(
+                            directive.module_id,
+                            loc.kind,
+                            Some(loc.def.krate),
+                        ));
+                        return recollect_without(self);
+                    }
+
```

Comment:
> I changed the handling of disabled attribute macro expansion to
immediately ignore the macro and report an unresolved_proc_macro,
because otherwise they would now result in loud unresolved_macro_call
errors. I hope this doesn't break anything.

I'm not sure if this breaks anything or not, there is the comment on `reseeed_with_unresolved_attribute`:
```
    /// When the fixed-point loop reaches a stable state, we might still have some unresolved
    /// attributes (or unexpanded attribute proc macros) left over. This takes one of them, and
    /// feeds the item it's applied to back into name resolution.
    ///
    /// This effectively ignores the fact that the macro is there and just treats the items as
    /// normal code.
    ///
    /// This improves UX when proc macros are turned off or don't work, and replicates the behavior
    /// before we supported proc. attribute macros.
```

Reply:
Ah yeah, I should at least update that comment. IMO it shouldn't change anything; if attribute expansion is disabled, known proc macro attributes will just get 'reseeded' immediately instead of after going through everything else.

We already do the same when proc macros are enabled, but the proc macro expander can't be found (with the `is_dummy` check further down).